group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1307,13 +1307,14 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
             }
         }
 
+        // The subscription metadata is updated in two cases:
+        // 1) The member has updated its subscriptions;
+        // 2) The refresh deadline has been reached.
```

Comment:
nit: Could we move it to right before `subscribedTopicNamesMap = group.computeSubscribedTopicNames(member, updatedMember);`?

Reply:
I thought you wanted it above the if statement in the last comment haha, I think it was there originally